aten/src/ATen/TensorImpl.h

```diff
@@ -185,22 +215,31 @@ struct AT_API TensorImpl : public c10::intrusive_ptr_target {
     return n;
   }
   bool compute_contiguous() const;
-
- protected:
   void refresh_numel() {
     numel_ = compute_numel();
   }
   void refresh_contiguous() {
     is_contiguous_ = compute_contiguous();
   }
+
+ protected:
   TensorTypeId type_id_;
   // INVARIANT: When storage is non-null, this scalar type must
   // agree with the scalar type in storage
   ScalarType scalar_type_;
   bool is_variable_ = false;
   bool is_wrapped_number_ = false;
+  bool has_storage_ = false;
+  bool has_strides_ = false;
+  bool support_resize_by_maybe_zero_dim_ = false;
```

Comment:
`has_storage_` and `has_strides_` are self explanatory, but `support_resize_by_maybe_zero_dim_` is not; worth a comment saying what it is about.